src/lib.rs

```diff
@@ -203,23 +204,125 @@ impl From<std::io::ErrorKind> for Error {
     }
 }
 
+bitflags! {
+    /// Network device flags.
+    pub struct IfFlags: u32 {
+        /// Set if the device is a loopback interface
+        const LOOPBACK = raw::PCAP_IF_LOOPBACK;
+        /// Set if the device is up
+        const UP = raw::PCAP_IF_UP;
+        /// Set if the device is running
+        const RUNNING = raw::PCAP_IF_RUNNING;
+        /// Set if the device is a wireless interface; this includes IrDA as well as radio-based
+        /// networks such as IEEE 802.15.4 and IEEE 802.11, so it doesn't just mean Wi-Fi
+        const WIRELESS = raw::PCAP_IF_WIRELESS;
+    }
+}
+
+impl From<u32> for IfFlags {
+    fn from(flags: u32) -> Self {
+        IfFlags::from_bits_truncate(flags)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Indication of whether the adapter is connected or not; for wireless interfaces, "connected"
+/// means "associated with a network".
+pub enum ConnectionStatus {
+    /// It's unknown whether the adapter is connected or not
+    Unknown,
+    /// The adapter is connected
+    Connected,
+    /// The adapter is disconnected
+    Disconnected,
+    /// The notion of "connected" and "disconnected" don't apply to this interface; for example, it
+    /// doesn't apply to a loopback device
+    NotApplicable,
+}
+
+impl From<u32> for ConnectionStatus {
+    fn from(flags: u32) -> Self {
+        match flags & raw::PCAP_IF_CONNECTION_STATUS {
+            raw::PCAP_IF_CONNECTION_STATUS_UNKNOWN => ConnectionStatus::Unknown,
+            raw::PCAP_IF_CONNECTION_STATUS_CONNECTED => ConnectionStatus::Connected,
+            raw::PCAP_IF_CONNECTION_STATUS_DISCONNECTED => ConnectionStatus::Disconnected,
+            raw::PCAP_IF_CONNECTION_STATUS_NOT_APPLICABLE => ConnectionStatus::NotApplicable,
+            // DeviceFlags::CONNECTION_STATUS should be a 2-bit mask which means that the four
+            // values should cover all the possibilities.
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct DeviceFlags {
+    pub if_flags: IfFlags,
```

Comment:
here I for sure want `flags:`

Reply:
That would potentially cause confusion with the `flags` field in `Device` leading to code like `device.flags.flags` which may be confusing. Happy to rename though, so open to other suggestions.

Reply:
we could choice to remove this intermediate struct and just put the flags and the connection status directly in Device.

Reply:
That was what I originally did. See the code here: https://github.com/rust-pcap/pcap/pull/252/files/606bdc33228ee8a386e739192837c14efb05a1d5. However, I do find that a bit uglier as pcap's `flags` contain two separate bits of information:
- the interface flags (LOOPBACK, UP, RUNNING, WIRELESS)
- connection status (which is not a flag but an enum of 4 values behind a bitmask)

I found the mix of these two a bit problematic and abstraction-wise I think separating them in wrapper struct is the `pcap` way of exposing nicer abstractions.

Reply:
The problem with the above approach is that `CONNECTION_STATUS` is not a flag, but a bitmask which made the unified struct internally inconsistent

Reply:
bitflags can easy have flags that are not only one bit, if you wish you could remove the enum and put all in the bitflags struct.